spec/mongo/collection_spec.rb

```diff
@@ -2000,6 +2000,51 @@ def generate
   end
 
   describe '#count_documents' do
+
+        context 'no argument provided' do
+
+          # no arguments
+          let(:view) do
+            described_class.new(authorized_collection, {}, {})
+          end 
+
+          before do
+            authorized_collection.delete_many
+          end
+
+          context 'empty collection' do
+            it 'returns true' do
+              expect(view.count_documents()).to eq(view.count_documents({}))
+            end
```

Comment:
I definitely see the reasoning behind this test, but I think there is a potential pitfall -- what if the behavior of `count_documents` breaks? i.e. what if `view.count_documents()` and `view.count_documents({})` both return the same wrong thing? This test would still pass but the driver would be broken.

I would test this method against what you expect it to return, and modify the test description (currently `it "returns true"`) accordingly.